### ISSUE TITLE
Spinner library panics often

### ISSUE THREAD

**Issue by brooksmtownsend** [wasmCloud/wash-archive#57]
The spinner library, while great in a normal execution environment, panics often depending on the terminal. I've observed the following error on GitHub Actions runs and on a Windows Powershell environment:
```
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: NotSupported', C:\Users\brooks\.cargo\registry\src\github.com-1ecc6299db9ec823\spinner-0.5.0\src\lib.rs:164:37
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

This needs more investigation to find out what's causing the panic. I already have disabled the spinner with JSON output, but I think it might be wise to catch errors stemming from instantiating the spinner: https://github.com/wasmCloud/wash/blob/main/src/ctl.rs#L794.

**Comment by brooksmtownsend** [comment]
Note: These panics do _not_ cause the program to fail, and the rest of the command will be executed as normal. For now, this does not break the cli, it just displays an unnecessary panic that can cause confusion

**Comment by brooksmtownsend** [comment]
Moving to `0.3.0` milestone as it does not need to be fixed before the `0.2.0` release since this bug does not affect functionality

**Comment by VestigeJ** [comment]
Spinners Depends on spinner^0.5.0. Spinner is currently a read only archived repo but [seems to be missing some error handling](https://github.com/TheNeikos/spinner/issues/22) so the bug remains in the Spinners crate for now.


https://github.com/TheNeikos/spinner/blob/ae9d25cb4690d5e17978db12fac7663156f2690f/src/lib.rs#L164

**Comment by stevelr** [comment]
In case it's useful, there's https://crates.io/crates/indicatif which seems to be actively maintained.

**Comment by brooksmtownsend** [comment]
`indicatif` looks like a great option, when replacing spinners we'll just have to ensure a few scenarios work well:
- Normal progress bar in Text output mode
- No progress bar in JSON output mode
- Works well in a non-interactive terminal such as the Github actions runner (e.g. doesn't panic when a terminal is non-interactive)
- Functions in windows Powershell, which the current spinner library does not.

We can test the Github Actions runner by simply including a test with `indicatif`, and I'd be happy to test this on Windows to ensure it works in Powershell.

**Comment by byblakeorriver** [comment]
Currently working on this one: https://github.com/byblakeorriver/wash/tree/issue_57
Will open a PR when I am far enough along.

**Comment by stevelr** [comment]
a windows 10 powershell user reported this today with wash 0.7.2.

**Comment by byblakeorriver** [comment]
I am getting close with the transition to `indicatif`. However, I do not have a Windows machine to test on.

**Comment by brooksmtownsend** [comment]
@byblakeorriver Feel free to open a PR and denote that, we have tests running on windows in github and I have a windows machine that I can assist with

**Comment by brooksmtownsend** [comment]
Closed by #246
